From 5fac0b8dcb3770c25b27fb5454cfb01e00fdefc3 Mon Sep 17 00:00:00 2001
From: Vikash Bansal <bvikas@vmware.com>
Date: Mon, 25 Jan 2021 11:08:19 +0530
Subject: [PATCH 6/7] crypto: AF_ALG - eliminate code duplication

The handling function for setsockopt contains duplicated code which is
cleaned up with this patch. This patch does not change the functionality.

Signed-off-by: Stephan Mueller <smueller@chronox.de>
Signed-off-by: Vikash Bansal <bvikas@vmware.com>
Signed-off-by: Shivani Agarwal <shivani.agarwal@broadcom.com>
---
 crypto/af_alg.c | 19 +++----------------
 1 file changed, 3 insertions(+), 16 deletions(-)

diff --git a/crypto/af_alg.c b/crypto/af_alg.c
index 6ea2e3a48..9c7653dd3 100644
--- a/crypto/af_alg.c
+++ b/crypto/af_alg.c
@@ -376,45 +376,32 @@ static int alg_setsockopt(struct socket *sock, int level, int optname,
 	if (level != SOL_ALG || !type)
 		goto unlock;
 
+	if (sock->state == SS_CONNECTED)
+		goto unlock;
+
 	switch (optname) {
 	case ALG_SET_KEY:
 	case ALG_SET_KEY_BY_KEY_SERIAL:
-		if (sock->state == SS_CONNECTED)
-			goto unlock;
-
 		if (optname == ALG_SET_KEY_BY_KEY_SERIAL)
 			err = alg_setkey_by_key_serial(ask, optval, optlen);
 		else
 			err = alg_setkey(sk, optval, optlen, type->setkey);
 		break;
 	case ALG_SET_PUBKEY:
-		if (sock->state == SS_CONNECTED)
-                        goto unlock;
-
 		err = alg_setkey(sk, optval, optlen, type->setpubkey);
 		break;
 	case ALG_SET_DH_PARAMETERS:
-		if (sock->state == SS_CONNECTED)
-			goto unlock;
-
 		err = alg_setkey(sk, optval, optlen, type->dhparams);
 		break;
 	case ALG_SET_ECDH_CURVE:
-		if (sock->state == SS_CONNECTED)
-			goto unlock;
-
 		err = alg_setkey(sk, optval, optlen, type->ecdhcurve);
 		break;
 	case ALG_SET_AEAD_AUTHSIZE:
-		if (sock->state == SS_CONNECTED)
-			goto unlock;
 		if (!type->setauthsize)
 			goto unlock;
 		err = type->setauthsize(ask->private, optlen);
 		break;
 	case ALG_SET_DRBG_ENTROPY:
-		if (sock->state == SS_CONNECTED)
-			goto unlock;
 		if (!type->setentropy)
 			goto unlock;
 
-- 
2.39.4

